### DISCUSSION TITLE
Get positive number when computing the variational expectations.

### DISCUSSION THREAD

**Discussion by shixinxing** [GPflow/GPflow#1691] [General]
I use **gpflow.likelihood.variational_expectations** to compute one of my ELBO terms: ***Eq(f)[log(p(y|f))]***.

Specifically, 

- q(f) is a Gaussian distribution with mean ***Fmu*** and variance ***Fvar*** ;
- p(y|f) is a Gaussian with mean ***f*** and variance ***variance***;

Here is the source code for Gaussian likelihood: 

> def _variational_expectations(self, Fmu, Fvar, Y):
        return tf.reduce_sum(
            -0.5 * np.log(2 * np.pi)
            - 0.5 * tf.math.log(self.variance)
            - 0.5 * ((Y - Fmu) ** 2 + Fvar) / self.variance,
            axis=-1,
        )

theoretically, the above term should be always negative, no matter how to choose these parameters like ***Fmu***, ***Fvar*** and ***variance***.

**However, I find this term sometimes goes to positive when the ***variance*** is set to a relative small number (about 0.003, but not too small).**

For example, set **Fmu=Y, Fvar=0.001**, I plot the results when variance is between **0.001 and 0.2**:
![plot](https://user-images.githubusercontent.com/47957281/121010194-af654100-c7c7-11eb-8cc4-8e663a8f0476.png)

It seems that the term ***-0.5 * tf.math.log(variance)*** is too positive or to refer to some problem for numerical stability...
Is there any solution to fix this ?
Or is this just a common phenomenon...?

**Comment by st--** [comment]
> theoretically, the above term should be always negative, no matter how to choose these parameters like Fmu, Fvar and variance.

What's the reason that this term would have to be negative? p(y | f) is a density, not a probability, so it'd be possible for it to be larger than 1, and hence for log p(y | f) to be larger than zero.

**Reply by shixinxing** [reply]
Yes, you are right. I realized where I was wrong several months ago. Thank you again!
